tests/test_cli_compile.py

```diff
@@ -511,6 +511,64 @@ def test_editable_package_vcs(runner):
     assert "click" in out.stderr  # dependency of pip-tools
 
 
+@pytest.mark.network
+def test_compile_cached_vcs_package(runner, tmp_path):
+    """
+    Test pip-compile doesn't write local paths for cached wheels of VCS packages.
+
+    Regression test for issue GH-1647.
+    """
+    vcs_package = (
+        "typing-extensions @ git+https://github.com/python/typing_extensions@"
+        "9c0759a260fe126210a1e2026720000a3c40a919"
+    )
+    vcs_wheel_prefix = "typing_extensions-4.3.0-py3"
+
+    # Install and cache VCS package.
+    venv_path = tmp_path / "venv"
+    subprocess.run(
+        [
+            sys.executable,
+            "-m" "venv",
+            venv_path,
+        ],
+        check=True,
+    )
+    subprocess.run(
+        [
+            venv_path / "bin" / "python",
+            "-m" "pip",
+            "install",
+            vcs_package,
+        ],
+        check=True,
+    )
+    assert (
+        vcs_wheel_prefix
+        in subprocess.run(
+            [
+                sys.executable,
+                "-m" "pip",
+                "cache",
+                "list",
+                "--format=abspath",
+                vcs_wheel_prefix,
+            ],
+            check=True,
+            capture_output=True,
+            text=True,
+        ).stdout
+    )
+
+    with open("requirements.in", "w") as req_in:
```

Comment:
Why is this written to the project root?

Reply:
Not sure. I just followed the pattern used already for the other tests.